core/src/main/java/org/web3j/protocol/websocket/WebSocketService.java

```diff
@@ -106,21 +111,30 @@ private void connectToWebSocket() throws InterruptedException, ConnectException
         }
     }
 
-    private void setWebSocketListener() {
+    private void setWebSocketListener(Consumer<String> onMessage, Consumer<Throwable> onError, Runnable onClose) {
         webSocketClient.setListener(new WebSocketListener() {
```

Comment:
@zacscoding , use of `Runnable` can be confusing as it would imply the implementation is invoked by another thread. Could you instead use `Consumer<Void>` please?
@iikirilov , thoughts?

Reply:
If use Consumer<Void>, then have to called like onClose.accept(null);
is ok ?

Reply:
yes